lib/flowcommerce_spree/webhook_service.rb

```diff
@@ -113,6 +113,7 @@ def hook_order_upserted_v2
 
         order.update_columns(attrs_to_update)
         order.create_tax_charge! if flow_data_submitted
+        order.persist_flow_addresses
```

Comment:
I think this line is better to move before line 114, updating the `attrs_to_update` hash - thus, doing the order update in one DB query.

Reply:
oh you mean like the `persist_flow_address` is not the one in charge saving the address, but instead returning a hash with the IDs of the ship and bill address and add it to the attrs_to_update?

Reply:
Yep, exactly!

Reply:
Looks like this is not going to work. While testing that approach I have realized that the order's flow information is not persisted within the meta column. And therefore the flow_ship_address and flow_bill_address return nil values. It does work if called multiple times, though. 

I believe I will go with a middle approach. I will configure the `persist_flow_address` method to update both ship_address_id and bill_address_id in one unique DB call. It is not perfect but at least will be clearer.

Reply:
Strange - after this line the order meta should contain all the info received on the webhook event - https://github.com/mejuri-inc/flowcommerce_spree/blob/a0f68b6d401a95c96be06cc3c1432de95f9f04f5/lib/flowcommerce_spree/webhook_service.rb#L100

Reply:
Yep! Moving at the bottom the persist_address call works like a charm. I will try to improve it !

Reply:
Ended up moving it back up! The problem was that the order's status was not set yet. So I had to change it to validate either the order.complete? or the attrs_to_update[:state] (since the attrs_to_update[:state] is not set after the order is set as completed).